### DISCUSSION TITLE
Using cookies in curl

### DISCUSSION THREAD

**Discussion by JakeElwood777** [curl/curl#12954] [Q&A] [answered]
Hello community,

i'm using the c-sourcesode of curl to query internetsites. In general is works fine with my code. But on this site i got an error: https://www.gametracker.com/. The message is : "Sorry, you have been blocked". When opening the website in my edge browser all works fine. I think it has something to do with the cookies settings, but i'm not sure. Here is my code:

`
boolean CurlEasy_DownloadiNetFileToBuffer(char *pURL, char *pUserAgent, char **recieve_buffer, u_int *len_recievebuffer, int timout_ms){

	CURL				*curl;
	CURLcode			res;
	easy_curl_buff_t			eabf = {0};
	boolean				back = FALSE;

	curl_global_init(CURL_GLOBAL_DEFAULT);
	curl = curl_easy_init();
	if(!curl){
		return FALSE;
	}
	curl_easy_setopt(curl, CURLOPT_URL, pURL);
	curl_easy_setopt(curl, CURLOPT_USERAGENT, pUserAgent);
	curl_easy_setopt(curl, CURLOPT_CONNECTTIMEOUT_MS, timout_ms);
	curl_easy_setopt(curl, CURLOPT_WRITEDATA, &eabf);
	curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, CurlEasy_WriteToBuffer);
	curl_easy_setopt(curl, CURLOPT_COOKIEJAR, "cookies.txt");
	res = curl_easy_perform(curl);
	if(res == CURLE_OK){
		if(eabf.buffer){
			back = TRUE;
			eabf.buffer[eabf.len_buffer] = '\0';
			*recieve_buffer = eabf.buffer;
			*len_recievebuffer = (u_int)eabf.len_buffer;
		}
	}else{
		if(eabf.buffer){
			free(eabf.buffer);
		}
	}
	curl_easy_cleanup(curl);
	return back;
}

`
What i'm doing wrong? Do you have any idea? I hope you can help me here. Thanks so far.

JakeElwood777

**Comment by jay** [comment, marked as answer]
> But on this site i got an error: https://www.gametracker.com/. The message is : "Sorry, you have been blocked".

Yeah some websites do that, usually through CDNs most notably cloudflare which is very sensitive. You would need to contact the site owner to figure out why you've been blocked. For example it may have blocked your request due to the user agent and/or the frequency of requests.

**Comment by JakeElwood777** [comment]
Thank you for your advice. The provider is indeed cloudflare. I will ask the site owner why i am blocked. Maybe i can tehn fix it.

**Comment by JakeElwood777** [comment]
I got no answere from the website owner. Is there some kind of "curl-cloudflare" workaround/bypass to get it working. Has someone been successfull to curl cloudflare hosted/protected websites?
I uses the same useragent string like edge do at other http web sites. I recorded this with wireshark.

**Comment by dfandrich** [comment]
There's a project called [curl-impersonate](https://github.com/lwthiker/curl-impersonate) that tweaks curl so that it makes requests that look identical to how a browser makes them. If they're doing browser fingerprinting, that might help.
